### PR TITLE
Move native crash installer to core module

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Moves the native crash installer to the core module. This is a pre-requisite for a later refactor.

## Testing

Relied on existing test coverage.

### CODE REVIEW

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/injection/ModuleInitBootstrapper.kt [hunk 1]
```diff
@@ -350,10 +352,17 @@ internal class ModuleInitBootstrapper(
                     }
 
                     postInit(NativeFeatureModule::class) {
+                        val configService = configModule.configService
                         serviceRegistry.registerServices(
                             lazy { nativeFeatureModule.nativeThreadSamplerService }
                         )
-                        nativeFeatureModule.nativeCrashHandlerInstaller?.install()
+
+                        if (configService.autoDataCaptureBehavior.isNativeCrashCaptureEnabled()) {
```
Comment (priettt) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/injection/ModuleInitBootstrapper.kt:360]: Do we need to check this here? We're already checking the ndk config in the feature module. If it's not enabled, it'll return null.

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/injection/ModuleInitBootstrapper.kt [hunk 2]
```diff
@@ -350,10 +352,17 @@ internal class ModuleInitBootstrapper(
...
+                            val worker = workerThreadModule.backgroundWorker(Worker.Background.IoRegWorker)
+                            worker.submit {
```
Comment (priettt) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/injection/ModuleInitBootstrapper.kt:362]: We are passing a worker to the nativeCrashHandlerInstaller, and `install()` happens within that worker already.

File: embrace-android-features/src/test/java/io/embrace/android/embracesdk/internal/injection/NativeFeatureModuleImplTest.kt [hunk 3]
```diff
@@ -94,8 +88,6 @@ internal class NativeFeatureModuleImplTest {
         )
 
         module = createNativeFeatureModule(fakeConfigModule)
-
-        assertNotNull(module.nativeCrashHandlerInstaller)
```
Comment (priettt) [inline embrace-android-features/src/test/java/io/embrace/android/embracesdk/internal/injection/NativeFeatureModuleImplTest.kt:98]: could we move the native crash handler tests to the core module?

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/injection/ModuleInitBootstrapper.kt (hunk 1 above)
Comment (fractalwrench) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/injection/ModuleInitBootstrapper.kt:360]: Removed this redundant check

**Pull request merged**